q2_diversity/tests/test_alpha_rarefaction.py

```diff
@@ -0,0 +1,428 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2017, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import io
+import os
+import tempfile
+import unittest
+
+import biom
+import numpy as np
+import pandas.testing as pdt
+import qiime2
+import skbio
+import pandas as pd
+from q2_diversity import alpha_rarefaction
+from q2_diversity._alpha._visualizer import (
+    _compute_rarefaction_data, _compute_summary, _reindex_with_metadata,
+    _beta_rarefaction_jsonp)
+
+
+class AlphaRarefactionTests(unittest.TestCase):
```

Comment:
Can you add tests for:

- empty table raises an error
- metadata with a superset of the IDs found in feature table (those extra IDs are ignored)
- metadata with a subset of the IDs found in feature table (raise an error)